caddytls/ctmonitor.go

```diff
@@ -306,9 +306,11 @@ func prepQuery(domainName string, config QueryConfig) (query string) {
 	return query
 }
 
-func putLatestID(currentID int, fileName string) error {
-	writeValue := strconv.Itoa(currentID)
-	return ioutil.WriteFile(fileName, []byte(writeValue), 0600)
+func putLatestIndex(config QueryConfig) error {
+	//writeValue := strconv.Itoa(currentID)
+
+	jsonBlob := []byte(`{"jsonString":config.CTConfig.IncludeSubdomains. "jsonString":config.CTConfig.IncludeWildcards", "jsonString":config.CTConfig.LastID}`)
+	return ioutil.WriteFile(ctConfigFile, jsonBlob, 0600)
```

Comment:
This is not correct. :)

Use `json.NewEncoder(outFile).Encode(config)`, where `outFile` is the opened ctmonitor_config file.

Reply:
Here, I did something similar today:

```go
f, err := os.Create(ctConfigFile)
if err != nil {
	return fmt.Errorf("opening %s file for writing: %v", ctConfigFile)
}
defer f.Close()
err = json.NewEncoder(f).Encode(config)
if err != nil {
	return fmt.Errorf("writing CT monitor config: %v", err)
}
```

Simple, elegant.

Reply:
I like it